app/models/event.rb

```diff
@@ -0,0 +1,18 @@
+class Event < ApplicationRecord
+  validates :description, length: { in: 10..120 }
+  validates :location, length: { in: 5..50 }
+
+  belongs_to :creator, class_name: 'User'
+  has_many :invitations, dependent: :destroy
+
+  scope :upcoming, -> { where('date >= ?', DateTime.now.to_date) }
+  scope :past, -> { where('date < ?', DateTime.now.to_date) }
+
+  def event_attendees
+    users = []
+    invitations.each do |i|
+      users << User.find_by_id(i.user_id)
+    end
+    users
+  end
```

Comment:
Event attendees should be retrieved by creating  a _has\_many through_ or _has\_and\_belongs\_to\_many_ association with your join model or event model respectively..

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/attendee_assoc_requirement.png)